addOns/zest/src/test/java/org/zaproxy/zap/extension/zest/ZestZapUtilsUnitTest.java

```diff
@@ -65,6 +109,35 @@ void shouldRemoveIgnoredHeadersIfNotIncludingAllWhenConvertingHttpMessageToZestR
         assertThat(zestRequest.getHeaders(), is(equalTo("A: 1\r\nHost: example.com\r\n")));
     }
 
+    @Test
+    void shouldIncludeResponseIfEnabledWhenConvertingHttpMessageToZestRequest() throws Exception {
+        // Given
+        ZestParam zestParam = createZestParam();
+        zestParam.setIncludeResponses(true);
+        HttpMessage httpMessage = createRequest("A: 1\r\nB: 2\r\nHost: example.com\r\n");
+        var response = "HTTP/1.1 200 OK\r\nC: 3\r\n\r\n";
+        httpMessage.setResponseHeader(response);
+        // When
+        ZestRequest zestRequest = ZestZapUtils.toZestRequest(httpMessage, false, false, zestParam);
+        // Then
+        assertThat(zestRequest.getResponse(), is(not(nullValue())));
+        assertThat(zestRequest.getResponse().getHeaders(), is(equalTo(response)));
```

Comment:
Doesn't this assertion imply the preceding one?

Reply:
Yes but better fail with a test assertion than NPE.

Reply:
Okay!